great_expectations/checkpoint/checkpoint.py

```diff
@@ -88,28 +104,111 @@ def _get_batches_to_validate(self, batches):
+                config = self.config
+            if isinstance(config, dict):
+                config = CheckpointConfig(**config)
+            template_name = runtime_kwargs.get("template_name") or config.template_name
+
+            if not template_name:
+                substituted_config = deepcopy(config)
+                if any(runtime_kwargs.values()):
+                    substituted_config.update(runtime_kwargs=runtime_kwargs)
+
+                self._substituted_config = substituted_config
+                return substituted_config
+            else:
+                template_config = self.data_context.get_checkpoint(
+                    checkpoint_name=template_name, return_config=True
+                )
 
-        return validators
+                if float(template_config.config_version) != float(
+                    config.config_version
+                ):
+                    raise CheckpointError(
+                        f"Invalid template '{template_name}' (ver. {template_config.config_version}) for Checkpoint "
+                        f"'{config}' (ver. {config.config_version}. Checkpoints can only use templates with the same config_version."
+                    )
+
+                if template_config.template_name is not None:
+                    substituted_config = self.get_substituted_config(
+                        config=template_config
+                    )
+                else:
+                    substituted_config = template_config
+
+                # merge template with config
+                substituted_config.update(
+                    other_config=config, runtime_kwargs=runtime_kwargs
+                )
+                self._substituted_config = substituted_config
+                return substituted_config
+
+    def run(
+        self,
+        template_name: Optional[str] = None,
+        run_name_template: Optional[str] = None,
+        expectation_suite_name: Optional[str] = None,
+        batch_request: Optional[Union[BatchRequest, dict]] = None,
+        action_list: Optional[List[dict]] = None,
+        evaluation_parameters: Optional[dict] = None,
+        runtime_configuration: Optional[dict] = None,
+        validations: Optional[List[dict]] = None,
+        profilers: Optional[List[dict]] = None,
+        run_id=None,
+        run_name=None,
+        run_time=None,
+        result_format=None,
+        **kwargs,
+    ):
+        runtime_kwargs = {
+            "template_name": template_name,
+            "run_name_template": run_name_template,
+            "expectation_suite_name": expectation_suite_name,
+            "batch_request": batch_request,
+            "action_list": action_list,
+            "evaluation_parameters": evaluation_parameters,
+            "runtime_configuration": runtime_configuration,
+            "validations": validations,
+            "profilers": profilers,
+        }
+        substituted_config = self.get_substituted_config(runtime_kwargs=runtime_kwargs)
```

Comment:
This looks like an abrupt ending to the `run` method -- isn't there anything more after line 214?  Or is this just GitHub now showing me more?..

Reply:
Sorry, I should have added a TODO comment. Will finish up run method next